polars/polars-lazy/polars-plan/src/logical_plan/optimizer/fused.rs

```diff
@@ -35,7 +35,12 @@ fn check_eligible(
         .get(*right)
         .get_type(&schema, Context::Default, expr_arena)?;
     let type_left = &field_left.dtype;
-    if type_left.is_numeric() && type_right.is_numeric() {
+    // Exclude literals for now as these will not result in fused operation downstream
```

Comment:
```suggestion
    // Exclude literals for now as these will not benefit from fused operations downstream #9857
    // This optimization would also interfere with the `col -> lit` type-coercion rules
    // And it might also interfere with constant folding which is a more suitable optimizations here
```